src/main/java/com/helger/smpmate/util/Tokenizer.java

```diff
@@ -0,0 +1,121 @@
+package com.helger.smpmate.util;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.*;
+
+/**
+ * Diese Klasse ist eine Abwandlung von {@link StringTokenizer}, die mehr den Erwartungen entspricht, wenn die
+ * Trennzeichen keine Whitespaces sind. Die Klasse {@link StringTokenizer} bündelt benachbarte Trennzeichen und macht
+ * sie dadurch für CSV-Belange unbrauchbar.
+ * Diese Klasse hingegen behandelt jedes Trennzeichen als LF (linefeed) und erfasst jedes Token quasi als Zeile
+ * einer Testdatei.
+ */
+public class Tokenizer extends StringTokenizer implements Iterable<String> {
+    private final List<String> tokens;
+    private int pos;
+
+    public Tokenizer(String str) {
+        this(str, ",");
+    }
+
+    public Tokenizer(String str, String delim) {
+        super(str);
+
+        int n = str.length();
+
+        StringBuilder builder = new StringBuilder(str);
+
+        for (int i = 0; i < n; i++) {
+            if (delim.indexOf(builder.charAt(i)) >= 0) {
+                builder.setCharAt(i, '\n');
+            }
+        }
+
+        if (n > 0 && delim.indexOf(str.charAt(n - 1)) >= 0) {
+            builder.append('\n');
+        }
+
+        tokens = new ArrayList<>();
+
+        BufferedReader reader = new BufferedReader(new StringReader(builder.toString()));
+        try {
+            for (String line; (line = reader.readLine()) != null;) {
+                tokens.add(line);
+            }
+            reader.close();
+        }
+        catch (IOException e) {
+            throw new Error(e);
+        }
+
+        pos = 0;
+    }
+
+    public String[] tokens() {
+        return tokens.toArray(new String[0]);
```

Comment:
Meine IDE sagt dazu:

Call to 'toArray()' with pre-sized array argument 'new String[tokens.size()]' 
 Inspection info: Reports Collection.toArray() calls that are not in the preferred style, and suggests applying the preferred style.
There are two styles to convert a collection to an array:
A pre-sized array, for example, c.toArray(new String[c.size()])
An empty array, for example, c.toArray(new String[0])
In older Java versions, using a pre-sized array was recommended, as the reflection call necessary to create an array of proper size was quite slow.
However, since late updates of OpenJDK 6, this call was intrinsified, making the performance of the empty array version the same, and sometimes even better, compared to the pre-sized version. Also, passing a pre-sized array is dangerous for a concurrent or synchronized collection as a data race is possible between the size and toArray calls. This may result in extra nulls at the end of the array if the collection was concurrently shrunk during the operation.

Reply:
Na gut - dann halt nicht :)